### ISSUE TITLE
Enabling OPENSSL_NO_EC2M will cause build failure: /x86_64-pc-linux-gnu/bin/ld: ./libcrypto.so: undefined reference to `EC_POINT_get_affine_coordinates_GF2m'

### ISSUE THREAD

**Issue by Whissi** [openssl/openssl#9872]
```bash
$ mkdir /tmp/no-ec2m-test
$ cd /tmp/no-ec2m-test
$ git clone https://github.com/openssl/openssl.git
$ cd openssl
$ git checkout OpenSSL_1_1_0l
$ ./Configure linux-x86_64 enable-ec no-ec2m
$ make
```
This will fail with
```
[...]
make[2]: Entering directory '/tmp/no-ec2m-test/openssl'
LD_LIBRARY_PATH=.: gcc -DDSO_DLFCN -DHAVE_DLFCN_H -DNDEBUG -DOPENSSL_THREADS -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DRC4_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPADLOCK_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-1.1\"" -Wall -O3 -pthread -m64 -DL_ENDIAN -o apps/openssl apps/app_rand.o apps/apps.o apps/asn1pars.o apps/ca.o apps/ciphers.o apps/cms.o apps/crl.o apps/crl2p7.o apps/dgst.o apps/dhparam.o apps/dsa.o apps/dsaparam.o apps/ec.o apps/ecparam.o apps/enc.o apps/engine.o apps/errstr.o apps/gendsa.o apps/genpkey.o apps/genrsa.o apps/nseq.o apps/ocsp.o apps/openssl.o apps/opt.o apps/passwd.o apps/pkcs12.o apps/pkcs7.o apps/pkcs8.o apps/pkey.o apps/pkeyparam.o apps/pkeyutl.o apps/prime.o apps/rand.o apps/rehash.o apps/req.o apps/rsa.o apps/rsautl.o apps/s_cb.o apps/s_client.o apps/s_server.o apps/s_socket.o apps/s_time.o apps/sess_id.o apps/smime.o apps/speed.o apps/spkac.o apps/srp.o apps/ts.o apps/verify.o apps/version.o apps/x509.o -L. -lssl -L. -lcrypto -ldl -pthread
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./libcrypto.so: undefined reference to `EC_POINT_get_affine_coordinates_GF2m'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile.shared:179: link_app.gnu] Error 1
make[2]: Leaving directory '/tmp/no-ec2m-test/openssl'
make[1]: *** [Makefile:5976: apps/openssl] Error 2
make[1]: Leaving directory '/tmp/no-ec2m-test/openssl'
make: *** [Makefile:132: all] Error 2
```
Looks like there is something wrong with the `#ifndef` clause in https://github.com/openssl/openssl/blob/OpenSSL_1_1_0l/crypto/ec/ec_curve.c#L3193 and later from https://github.com/openssl/openssl/commit/d4a5dac9f9242c580fb9d0a4389440eccd3494a7

Same for OpenSSL 1.0.2t.

**Comment by romen** [comment]
Thanks for reporting this @Whissi 

I can confirm that indeed in the 1.1.0 this slipped in! But the mistake is not in 1.0.2 as I catched it in time (https://github.com/openssl/openssl/pull/9811#pullrequestreview-285600690)

I checked if I had the mistake in 1.10 but must have messed up, probably did not refresh my editor cache and checked there was no problem in 1.1.0 while still looking at the 1.0.2 fixed code.

I will open a PR to fix this bug, but 1.1.0 is officially EOL.

**Comment by Whissi** [comment]
Thanks.

Yes, the 1.0.2t problem we are seeing seems to be caused by something else. Will keep investigating.

**Comment by romen** [comment]
The fix is in #9873

**Comment by romen** [comment]
The fix is in a3b54f0f5de1ad17889fd23aee7c230eefc300cd, that has been merged through #9873 in the `OpenSSL_1_1_0-stable` branch.

The caveat is, quoting @mattcaswell that

> 1.1.0 is now EOL and we're never going to make another release - although I guess its possible people may still want to grab this patch from git.


Thanks again @Whissi for reporting this!
